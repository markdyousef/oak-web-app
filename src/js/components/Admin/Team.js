// @flow
import React, { PropTypes } from 'react';
import styled from 'styled-components';
import colors from '../../styles/colors';

const Container = styled.button `
    width: 100%;
    max-width: 400px;
    height: 60px;
    background-color: ${colors.white};
    border: 1px solid ${colors.green};
    border-radius: 3px;
    font-size: 15px;
    margin: 5px;
    cursor: pointer;
`;


const Team = ({ team, onSelect }:Object) => {
    return (
        <Container onClick={onSelect}>
            {team.name}
        </Container>
    );
};

Team.propTypes = {
    team: PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
    }).isRequired,
    onSelect: PropTypes.func.isRequired
};

export default Team;
