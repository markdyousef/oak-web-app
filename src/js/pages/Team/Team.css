:root {
    --primary-color: #5ed892;
    --main-black-color: #131517;
    --light-black-color: #2d3438;
    --dark-gray-color: #737a80;
    --main-gray-color: #f2f2f2;
    --light-gray-color: #fafafa;
    --main-white-color: #fff;
}

.container {
    width: 100%;
    height: 100%;
}

.right {
    width: 100%;
}
