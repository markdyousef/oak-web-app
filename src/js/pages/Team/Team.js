// @flow
import React from 'react';
import styled from 'styled-components';
import SideNav from '../../containers/SideNavContainer';
import TopNav from '../../containers/TopNavContainer';

const Container = styled.div`
    display: flex;
    ${''/* overflow-x: hidden;
    overflow-y: hidden; */}
    flex-direction: column;
`;

const InnerContainer = styled.div`
    z-index: 0;
    ${''/* overflow-x: hidden;
    overflow-y: hidden; */}
    display: flex;
`;

const Main = styled.section`
    display: flex;
    flex-direction: column;
    flex-basis: 100%;
    position: relative;
    min-width: 1em;
    max-width: 100%;
`;

const MainInner = styled.div`
    display: block;
    ${''/* overflow-y: auto; */}
    ${''/* overflow-x: hidden; */}
`;

export default ({ children }:Object) => (
    <Container>
        <InnerContainer>
            <SideNav />
            <Main>
                <TopNav team />
                <MainInner>
                    {children}
                </MainInner>
            </Main>
        </InnerContainer>
    </Container>
);
