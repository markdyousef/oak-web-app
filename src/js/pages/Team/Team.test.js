import React from 'react';
import { shallow } from 'enzyme';
import Team from './Team';

it('renders correctly', () => {
    const props = {
        children: null
    }
    const shallowComponent = shallow(<Team {...props} />);
    expect(shallowComponent).toMatchSnapshot();
});
