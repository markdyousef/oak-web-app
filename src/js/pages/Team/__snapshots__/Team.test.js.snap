// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.div>
        <styled.section>
            <Unknown />
            <styled.div />
        </styled.section>
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.div>
            <styled.section>
                  <Unknown />
                  <styled.div />
            </styled.section>
      </styled.div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": null,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.div>
                    <styled.section>
                              <Unknown />
                              <styled.div />
                    </styled.section>
          </styled.div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.div>
                    <styled.section>
                              <Unknown />
                              <styled.div />
                    </styled.section>
          </styled.div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown />,
}
`;
